### PR TITLE
doc: fix rendering of code blocks in list

### PR DESCRIPTION
**Pull request by tniessen**
Without proper indentation, some paragraphs would be rendered as code blocks:

screenshot showing incorrect rendering

### CODE REVIEW

File: doc/setup.md [hunk 1]
```diff
@@ -32,51 +32,51 @@ To use **Node-API** in a native module:
...
-    "<!(node -p \"require('node-addon-api').targets\"):node_addon_api_maybe",
-  ],
-```
+     ```gyp
+       'dependencies': [
+         "<!(node -p \"require('node-addon-api').targets\"):node_addon_api_maybe",
+       ],
+     ```
 
   4. If you would like your native addon to support OSX, please also add the
```
Comment (legendecas) [inline doc/setup.md:59]:
```suggestion
  3. If you would like your native addon to support OSX, please also add the
```

File: doc/setup.md [hunk 2]
```diff
@@ -32,51 +32,51 @@ To use **Node-API** in a native module:
...
+         ['OS=="mac"', {
+             'cflags+': ['-fvisibility=hidden'],
+             'xcode_settings': {
+               'GCC_SYMBOLS_PRIVATE_EXTERN': 'YES', # -fvisibility=hidden
+             }
+         }]
+       ]
+       ```
 
   5. Include `napi.h` in the native module code.
```
Comment (legendecas) [inline doc/setup.md:73]:
```suggestion
  4. Include `napi.h` in the native module code.
```

**Review by legendecas — APPROVED** [review]
Thanks!

**Comment by tniessen** [comment]
CI keeps failing with `No module named 'distutils'`

**Comment by tniessen** [comment]
Likely due to https://github.com/nodejs/node-gyp/issues/2869. How do we upgrade node-gyp to pull in https://github.com/nodejs/node-gyp/pull/2923?

**Comment by legendecas** [comment]
Locking python version in GitHub actions: https://github.com/nodejs/node-addon-api/pull/1403

**Comment by legendecas** [comment]
node-gyp is bundled with npm. npm v10 dropped support for Node.js v14 and v16. So I'm going to simply lock the python version on GitHub actions, to avoid breakage in the future as well.

**Pull request merged**